starboard/nplb/posix_compliance/posix_file_get_info_test.cc

```diff
@@ -89,6 +85,73 @@ TEST(PosixFileGetInfoTest, WorksOnStaticContentFiles) {
   }
 }
 
+TEST(PosixFileGetInfoTest, WorksOnADirectory) {
+  char dir_template[] = "/tmp/fstat_test_dir.XXXXXX";
+  char* dir_path = mkdtemp(dir_template);
+  ASSERT_TRUE(dir_path != nullptr);
+
+  int fd = open(dir_path, O_RDONLY);
+  ASSERT_NE(fd, -1);
+
+  struct stat info;
+  EXPECT_EQ(fstat(fd, &info), 0);
+  EXPECT_TRUE(S_ISDIR(info.st_mode));
+
+  close(fd);
+  rmdir(dir_path);
+}
+
+TEST(PosixFileGetInfoTest, FollowsSymbolicLink) {
+  ScopedRandomFile target_file(128);
+  std::string target_path = target_file.filename();
+
+  char dir_template[] = "/tmp/fstat_test_dir.XXXXXX";
+  char* dir_path = mkdtemp(dir_template);
+  ASSERT_TRUE(dir_path != nullptr);
+
+  std::string link_path = std::string(dir_path) + "/symlink";
+  ASSERT_EQ(symlink(target_path.c_str(), link_path.c_str()), 0);
+
+  // Open the symbolic link.
+  int fd = open(link_path.c_str(), O_RDONLY);
+  ASSERT_NE(fd, -1);
+
+  // fstat should report info about the target file, not the link.
+  struct stat info;
+  EXPECT_EQ(fstat(fd, &info), 0);
+  EXPECT_EQ(info.st_size, 128);        // Size should be target's size.
```

Comment:
Consider making the size a variable in this test so that its value is declared in a single place